source/extensions/filters/http/oauth/filter.h

```diff
@@ -0,0 +1,220 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "absl/strings/string_view.h"
+
+#include "common/common/assert.h"
+#include "common/config/datasource.h"
+#include "common/http/rest_api_fetcher.h"
+
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/server/filter_config.h"
+
+#include "extensions/filters/http/common/pass_through_filter.h"
+
+#include "envoy/extensions/filters/http/oauth/v3/oauth.pb.h"
+#include "extensions/filters/http/oauth/oauth.h"
+#include "extensions/filters/http/oauth/oauth_client.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Oauth {
+
+class OAuth2Client;
+
+// Helper class used to fetch secrets (usually from SDS).
```

Comment:
I imagine this is done to allow for pluggable secrets from out of tree? It would be nice if this file just used the `GenericSecretConfigProvier` directly, but maybe that's hard to reconcile with your internal usage?

Reply:
I think this is because `GenericSecretConfigProvier` only provide `DataSource` but not any structured data. We might want to evolve from there (e.g. support Any protobuf from there)

Reply:
Do we want to do that now or in the next iteration?